Extractor/NodeSources/NodeSetSource.cs

```diff
@@ -330,9 +335,54 @@ public void BuildNodes(IEnumerable<NodeId> rootNodes, bool isFullBrowse)
                 InitNodeState(update, node);
             }
 
+        }
+
+        public override async Task<NodeSourceResult?> ParseResults(CancellationToken token)
+        {
+            if (!NodeMap.Any()) return null;
+
+            FinalDestinationObjects.Clear();
```

Comment:
Should we implement locks on these objects? They are referenced from a lot of places.

Reply:
They are protected, and I don't see any parallelism in this class, it shouldn't be necessary?

Reply:
Should be fine then 👍🏼